src/plugins/data/server/search/collectors/usage.ts

```diff
@@ -38,44 +40,53 @@ export interface SearchUsage {
   trackSuccess(duration: number): Promise<void>;
 }
 
-export function usageProvider(core: CoreSetup): SearchUsage {
+export function usageProvider(
+  core: CoreSetup,
+  initializerContext: PluginInitializerContext<ConfigSchema>
+): SearchUsage {
   const getTracker = (eventType: keyof Usage) => {
     return async (duration?: number) => {
-      const repository = await core
-        .getStartServices()
-        .then(([coreStart]) => coreStart.savedObjects.createInternalRepository());
+      const config = await initializerContext.config
+        .create<ConfigSchema>()
+        .pipe(first())
+        .toPromise();
+      if (config.search.usageTelemetry.enabled) {
```

Comment:
I haven't actually ran into the scenario since we always defined a default value will never be undefined (from what I can tell). 

Throughout the repo we are doing optional chaining. For example,
```
` value?.search?.aggs?.shardDelay?.enabled`
```

 Should we just be safe and do it here. Like
```
config?.search?.usageTelemetry?.enabled
```

Reply:
Thanks. updated.